### PR TITLE
Project files restructuring

### LINKED ISSUE #1444: Refactor project files for consistency
Audit all project router.js files so they follow the same pattern:

- each router file is located in a component
- each component router file contains routes only for the component
- component router files are in the `/component/client/lib/router.js` file
- router onBefore or other hooks are either in core, or in the specific router file where they are used

Additionally, refactor the `Settings` component into several separate components. Based on discussion with @brylie and @frenchbread.

### PR DESCRIPTION
**Pull request by jykae**
Closes #1444 

## Proposed changes
- overall audit for project structure consistency based on mutual discussions & issue description

### CODE REVIEW

**Comment by jykae** [comment]
@apinf/developers please review

File: documentation/lib/router.js [hunk 1]
```diff
@@ -0,0 +1,4 @@
+Router.route("apiDocumentationEditor", {
```
Comment (brylie) [inline documentation/lib/router.js]: This needs to go in the `client/lib` folder. It is currently just in `lib`.

**Comment by brylie** [comment]
`catalogue/client/lib/routes.js` should be renamed to `.../router.js` for consistency.

**Comment by brylie** [comment]
There are a couple of minor changes needed here. Be sure that all router files and paths are consistent. This is nearly ready to merge.

**Comment by jykae** [comment]
@brylie Requested changes done, also pulled develop and did same structure for codegenerator code. Anything else?

File: core/client/lib/router.js [hunk 2]
```diff
@@ -20,40 +33,36 @@ var redirectToDashboard = function () {
...
-
 // Router.onBeforeAction(redirectToSignIn, {
 //   except: ['home', 'atSignUp', 'forgotPwd', 'atSignOut', 'catalogue', 'viewApiBackend', 'search']
 // });
 
 Router.onBeforeAction(redirectToDashboard, {only: ['forgotPwd', 'signOut']});
 
-Router.onAfterAction(onAfterAction);
+Router.map(function() {
+  this.route("home", {
```
Comment (brylie) [inline core/client/lib/router.js]: Move the `home` route to a `router.js` file in the Home component.

File: core/client/lib/router.js [hunk 3]
```diff
@@ -20,40 +33,36 @@ var redirectToDashboard = function () {
...
+    path: "/",
+    layoutTemplate: "homeLayout"
+  });
+
+  this.route("accountsAdmin", {
```
Comment (brylie) [inline core/client/lib/router.js]: Move the `accountsAdmin` route to a `router.js` file in the Accounts component.

**Comment by brylie** [comment]
This is looking good. Thanks for the additional cleanup. Lets make the two remaining changes to the users and accounts component routes, and then merge this PR.

**Comment by jykae** [comment]
@brylie Did the requested changes + moving account, profile related under users component as discussed in the meeting.

**Comment by jykae** [comment]
@brylie Ready for review

**Comment by brylie** [comment]
Move Branding collection schema into `branding/collection/schema.js`, following our [collection/schema structure guidelines](https://github.com/apinf/api-umbrella-dashboard/blob/develop/CONTRIBUTING.md#collectionschema-structure).

**Comment by brylie** [comment]
Move Branding permissions into `/branding/collection/permissions.js`

**Comment by brylie** [comment]
Change the Branding collection to use export syntax, as per our [collection guidelinets](https://github.com/apinf/api-umbrella-dashboard/blob/develop/CONTRIBUTING.md#collectionschema-structure).

**Comment by brylie** [comment]
Move publications to `.../collection/server/publications.js`, since they are directly related to collections.

File: branding/client/lib/router.js [hunk 4]
```diff
@@ -0,0 +1,5 @@
+Router.route('/settings/branding', {
+  name: 'settings.branding',
```
Comment (brylie) [inline branding/client/lib/router.js]: Rename this route to `branding`. Also, change the `pathFor` link in `/settings/client/menu/menu.html:6` to remove the `settings.` prefix from `route`.

**Comment by brylie** [comment]
Rename all `settings.` routes to remove the `settings.` prefix:

- `settings.profile` -> `profile`
- `settings.account` -> `account`
- `settings.branding` -> `branding`
- `settings.proxies` -> `proxies`

Also remember to update the `/settings/client/menu/menu.html` to fix the `pathFor route=` values to match the above changes.

**Comment by brylie** [comment]
Move `/users/lib/router.js` to `/users/client/lib/router.js`, for consistency with other router.js files in the project. This is because our router only works in client code, so does not need to be accessible to the server. Also, the `lib` prefix is necessary to make the router.js load before other files.

**Comment by brylie** [comment]
Move `/settings/lib/router.js` to `/settings/client/lib/router.js`

**Comment by brylie** [comment]
Move `/proxies/server/publications.js` to `/proxies/collection/server/publications.js`.

This is because publications are closely related to collections.

See: [project file structure guidelines](https://github.com/apinf/api-umbrella-dashboard/blob/develop/CONTRIBUTING.md#file-structure) for reference.

**Comment by brylie** [comment]
Move `/server/methods/apiUmbrella.js` to `/proxies/server/methods.js`.

**Comment by brylie** [comment]
Check with @frenchbread regarding the `/settings/server/methods.js:4`. We should be moving any API Umbrella related code out of the Settings module/collection. We need to figure out where the `getApiUmbrellaHostName` method should go.

**Comment by brylie** [comment]
Should we rename `/users/lib/accounts.js` to `/users/lib/config.js`?

**Comment by jykae** [comment]
@brylie @frenchbread Everything else except 2 last comments have been fixed as requested.

**Comment by brylie** [comment]
@jykae I went ahead and renamed the file. @frenchbread said that he has moved the method in another PR. Lets go ahead and merge these changes ASAP, as soon as you can review my modification.

**Pull request merged**